### DISCUSSION TITLE
How can I prevent Avalonia starting up with a transparent background window?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10687] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/30720036/225767367-e39e2c5e-73b4-45e3-92d9-c21352794b06.png)

Avalonia version: 11.0.0 Preview5
Operation System:Windows 10 22H2 19045.2728

When the Avalonia program starts up, it always renders the background transparent within one second then renders the right background color. This makes the user experience very bad.

No matter if I set the background of window with a color, it still renders it transparent when it starts up within one second.

I am not sure whether it is a bug or something I haven't configured rightly.

Could you please help me? Thank you.

**Comment by maxkatz6** [comment]
If you profile your application startup, most likely it is lagging at this point because of JIT.
[ReadyToRun](https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run) or [NativeAOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/) options should help with that noticeably.

**Reply by kyuranger** [reply]
> If you profile your application startup, most likely it is lagging at this point because of JIT. [ReadyToRun](https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run) or [NativeAOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/) options should help with that noticeably.

Before the program starts. In Initialize of App.axml.cs will load some image/localizer resource for theme. After release with AOT, it works better now. I tried to create a blank new project to test this. It still will render a transparent background but in a very short time.

**Comment by timunie** [comment, marked as answer]
@kyuranger also possible check if there is any heavy I/O task when you load the App. If you are loading some data, consider to use async/await to not block the UI from loading.

**Reply by kyuranger** [reply]
> @kyuranger also possible check if there is any heavy I/O task when you load the App. If you are loading some data, consider to use async/await to not block the UI from loading.

Actually, the program loads some image/localizer resources for the theme in Initialize of App.axml.cs. I can't use async/await or move the code anywhere.

**Reply by timunie** [reply]
> I can't use async/await or move the code anywhere.

why?! It should be possible tbh. Otherwise you have to live with what you have now. You can bind to Tasks. https://docs.avaloniaui.net/docs/data-binding/binding-to-tasks-and-observables

**Reply by kyuranger** [reply]
> > I can't use async/await or move the code anywhere.
> 
> why?! It should be possible tbh. Otherwise you have to live with what you have now. You can bind to Tasks. https://docs.avaloniaui.net/docs/data-binding/binding-to-tasks-and-observables

I use the resource like this:
`<Grid Background="{StaticResource  SecondaryBackground}"></Grid>`
After the MainWindow is loaded, it will load the resources by key immediately. If the resource is not loaded, VS will report that the resource is not exist. Because of this, I have to load the resources before MainWindow is loaded.

**Reply by timunie** [reply]
Use DynamicResource instead.

**Reply by kyuranger** [reply]
> Use DynamicResource instead.

Got it. Thank you so much.

**Reply by timunie** [reply]
Can this thread be marked as answered?

**Reply by kyuranger** [reply]
> > Use DynamicResource instead.
> 
> Got it. Thank you so much.

Yes. I marked it.
